test/end-to-end/pages/createImage.js

```diff
@@ -20,11 +20,11 @@ module.exports = class CreateImagePage extends MainPage {
     this.labelCreateImage = `${this.dialogRootElement} #myModalLabel`;
 
     // Blueprint Name label
-    this.labelBlueprintName = `${this.dialogRootElement} p[class="form-control-static"]`;
+    this.labelBlueprintName = `${this.dialogRootElement} #cmpsr-form-ctrl`;
```

Comment:
If we do use the the `data-id` attribute as shown below:
```
<p 
  className="form-control-static"
  data-id="cmpsr-form-blueprint"
>{this.props.blueprint.name}</p>
```
then we can't use the `#` with this attribute value as a selector.

The `#` would only precede the value provided for an `id` attribute. But most attributes would need to be identified using the [attribute css selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) as shown below:

```
this.labelBlueprintName = `${this.dialogRootElement} [data-id="cmpsr-form-blueprint"]`;
```

Reply:
Why does this one need to be identified using an attribute css selector while the other one is able to be referenced with `#`?

Reply:
I'm not really sure what the history behind this is, so I'm making up my own version here that hopefully clarifies this:

CSS was invented so that developers could assign styles to elements. Elements could be selected by their `id` attributes using `#`, by `class` attributes using `.`, or by elements themselves using their tag name (e.g. `p` or `div`). But then someone realized that we should be able to style things using any attribute, so the catch-all attribute selector was added to the spec (i.e. `[{attribute}]`). And since then, it's been agreed to that selecting elements by `id` for styling with css is bad practice.

We're using the CSS selector syntax here since the tests use the CSS selectors to identify elements. And in this case, `id` is ideal if it's present because it has to be unique. But it isn't always present. It's most likely going to be used for form elements, but it might not be present for other things like buttons and links.